### PR TITLE
Non lwtey

### PR DESCRIPTION
**Pull request by domsj**
This PR
- removes unneeded lwt-ness
- turn aSSert and aSSert_exists into queries
- fix a concurrency bug in confirm

### CODE REVIEW

File: src/client/client_protocol.ml [hunk 1]
```diff
@@ -145,46 +145,48 @@ let one_command stop (ic,oc,id) (backend:Backend.backend) =
   then
     Lwt.return true
   else
+    let wrap_exception f =
+      Lwt.catch
+        (fun () ->
+         try f ()
```
Comment (NicolasT) [inline src/client/client_protocol.ml]: I think this try-with is done by Lwt already?

File: src/client/client_protocol.ml [hunk 2]
```diff
@@ -492,123 +475,112 @@ let one_command stop (ic,oc,id) (backend:Backend.backend) =
...
           response_ok_unit oc )
-        (handle_exception oc)
     end
   | GET_KEY_COUNT ->
     begin
-      Lwt.catch
+      wrap_exception
         (fun() ->
            Logger.debug_f_ "connection=%s GET_KEY_COUNT" id >>= fun () ->
            backend # get_key_count () >>= fun kc ->
```
Comment (NicolasT) [inline src/client/client_protocol.ml]: I'm somewhat surprised `get` etc are non-Lwt, but `get_key_count` is.

File: src/client/client_protocol.ml [hunk 3]
```diff
@@ -492,123 +475,112 @@ let one_command stop (ic,oc,id) (backend:Backend.backend) =
...
     end
   | CONFIRM ->
     begin
       Llio.input_string ic >>= fun key ->
       Llio.input_string ic >>= fun value ->
-      Lwt.catch
+      wrap_exception
         (fun () ->
            Logger.debug_f_ "connection=%s CONFIRM: key=%S" id key >>= fun () ->
            backend # confirm key value >>= fun () ->
```
Comment (NicolasT) [inline src/client/client_protocol.ml:574]: I'm somewhat surprised `get` etc are non-Lwt, but `confirm` is.

File: src/client/client_protocol.ml (hunk 1 above)
Comment (domsj) [inline src/client/client_protocol.ml]: Hmm, apparently you're right

File: src/node/dump_store.ml [hunk 4]
```diff
@@ -21,10 +21,10 @@ open Interval
 
 module S = (val (Store.make_store_module (module Batched_store.Local_store)))
 
-let try_fetch name (f:unit -> 'a Lwt.t) (r2s: 'a -> string)  =
+let try_fetch name (f:unit -> 'a) (r2s: 'a -> string)  =
   Lwt.catch
     (fun () ->
-       f () >>= fun r ->
+       Lwt.wrap f >>= fun r ->
```
Comment (NicolasT) [inline src/node/dump_store.ml]: Is this `wrap` instead of simply calling?

Comment (domsj) [inline src/node/dump_store.ml]: I could write `let r = f () in` instead. I wrongly assumed the wrap was needed here, but Lwt.catch has an implicit wrap already...

File: src/client/client_protocol.ml (hunk 3 above)
Comment (domsj) [inline src/client/client_protocol.ml:574]: confirm might do an update, so this should remain Lwt.t

File: src/client/client_protocol.ml (hunk 2 above)
Comment (domsj) [inline src/client/client_protocol.ml]: hmm, somehow 'missed' this one

**Comment by domsj** [comment]
Updated, but I'll run CI again to be sure.

**Comment by domsj** [comment]
CI agrees with my latest changes here. Travis failure seems to be camltc related.

**Pull request merged**